.github/workflows/ci.yml

```diff
@@ -0,0 +1,94 @@
+name: CI
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  push:
+    branches:
+      - master
+env:
+  IMAGE_TAG: ${{ github.sha }}
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/cache@v1
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-yarn-client-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-client-
+      - name: yarn
+        run: yarn
+      - name: Lint
+        run: yarn lint
+      - name: Unit Test
+        run: yarn test
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/cache@v1
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-yarn-client-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-client-
+      - name: yarn
+        run: yarn
```

Comment:
what does this do before the Docker image creation? As far as I'm aware the `Dockerfile` should not be able to use local files such as `node_modules` at build time :thinking:

Reply:
I see, `COPY` takes the input of `dist/`. Reflecting on why I am not used to it.

Reply:
Some languages cannot do this, e.g. Python has files in the virtualenvs containing the dependencies that are not portable across different operating systems. I suppose you have nothing that gets compiled in `node_modules`?

Reply:
Nothing in node_modules should need compilation, there may be a post install step but it will only affect the files local to that dependency so the results should copy over also.

Reply:
If this were server side I would be a tad worried .... but given its building code to run across the world in peoples browers - mah! :+1:

Reply:
(it is called `node_modules` not `browser_modules`?)

Reply:
I think https://github.com/sass/node-sass/releases downloads a binary appropriate to the current OS rather than compiling, quite good.
Similar for https://github.com/giggio/node-chromedriver.

Reply:
The dependencies are being used by node but in the same environment to generate browser assets. As long as we're not changing `runs-on` between steps this should be fine. I've yet to come across a browser app thats had problems because it's been bundled on a particular OS.